lib/cache_mngr.c

```diff
@@ -156,10 +183,18 @@ int nl_cache_mngr_alloc(struct nl_sock *sk, int protocol, int flags,
 	if (flags & NL_ALLOCATED_SOCK)
 		BUG();
 
+	if(!sk_blocking) {
+		BUG();
+	}
+
 	mngr = calloc(1, sizeof(*mngr));
-	if (!mngr)
+	if (!mngr) {
+		// No manager yet, to free it in case of error
+		nl_socket_free(sk_blocking);
```

Comment:
this is inconsistent too what happens with `sk`. On failure, the sockets are left alone. The caller would need to handle them in that case. That is also missing in `nl_cache_mngr_alloc()` (but that code will go away).